server/etcdserver/apply/apply_kv_test.go

```diff
@@ -0,0 +1,408 @@
+		},
+	})
+	require.NoError(t, err)
+	require.False(t, response.Succeeded)
+	require.Len(t, response.Responses, 2)
+	responseRange, ok := response.Responses[0].Response.(*pb.ResponseOp_ResponseRange)
+	require.True(t, ok)
+	require.EqualValues(t, 2, responseRange.ResponseRange.Count)
+	responsePut, ok := response.Responses[1].Response.(*pb.ResponseOp_ResponsePut)
+	require.True(t, ok)
+	require.Nil(t, responsePut.ResponsePut.PrevKv)
+
+	// multiple key (create less/not equal) comparison (success), then delete(key)
+	response, _, err = appV3be.Txn(ctx, &pb.TxnRequest{
+		Compare: []*pb.Compare{
+			{
+				Result:      pb.Compare_LESS,
+				Target:      pb.Compare_CREATE,
+				Key:         fourKeys[0].Key,
+				TargetUnion: &pb.Compare_CreateRevision{CreateRevision: 5},
+			},
+			{
+				Result:      pb.Compare_NOT_EQUAL,
+				Target:      pb.Compare_VERSION,
+				Key:         fourKeys[1].Key,
+				TargetUnion: &pb.Compare_Version{Version: 3},
+			},
+		},
+		Success: []*pb.RequestOp{
+			{Request: &pb.RequestOp_RequestDeleteRange{
+				RequestDeleteRange: &pb.DeleteRangeRequest{
+					Key: fourKeys[0].Key,
+				},
+			}},
+		},
+	})
+	require.NoError(t, err)
+	require.True(t, response.Succeeded)
+	deleteResponse, ok := response.Responses[0].Response.(*pb.ResponseOp_ResponseDeleteRange)
+	require.True(t, ok)
+	require.EqualValues(t, 1, deleteResponse.ResponseDeleteRange.Deleted)
+}
+
+func TestBackendCompaction(t *testing.T) {
+	appV3be, _ := defaultApplierV3Backend(t)
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	// key(rev2) key2(rev3) key3(rev4,updated 6) key4(rev5, updated 7)
+	putKeys(ctx, t, fourKeys, appV3be)
+	putKeys(ctx, t, []pb.PutRequest{
+		{Key: []byte(key3), Value: []byte("temp3 value")},
+		{Key: []byte(key4), Value: []byte("temp4 value")},
+	}, appV3be)
+
+	_, _, _, err := appV3be.Compaction(&pb.CompactionRequest{Revision: 5})
+	require.NoError(t, err)
+
+	_, err = appV3be.Range(ctx, nil, &pb.RangeRequest{
+		Key:      []byte(key),
+		RangeEnd: zero,
+		Revision: 3,
+	})
+	require.EqualValues(t, mvcc.ErrCompacted, err)
+}
+
+func defaultQuotaApplierV3(t *testing.T) *quotaApplierV3 {
+	appV3be, be := defaultApplierV3Backend(t)
+
+	return &quotaApplierV3{
+		appV3be,
+		serverstorage.NewBackendQuota(appV3be.lg, TestingQuotaBackendBytes, be, "v3-applier"),
+	}
+}
+
+// `Put` beyond quota should yield "no space" error
+func TestQuotaPut(t *testing.T) {
+	qApp := defaultQuotaApplierV3(t)
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	// backend has only 128B of remaining space
+	_, _, err := qApp.Put(ctx, nil, &pb.PutRequest{
+		Key:   []byte(key),
+		Value: []byte(value128),
+	})
+	require.EqualValues(t, errors.ErrNoSpace, err)
+
+	// Putting 64B should go OK
+	result, _, err := qApp.Put(ctx, nil, &pb.PutRequest{
+		Key:   []byte(key),
+		Value: []byte(value64),
+	})
+	require.EqualValues(t, errors.ErrNoSpace, err)
```

Comment:
This assertion fails the test.

Reply:
The quota tests are sure to be reviewed. Beyond some leftovers, trusting exact DB startup sizes will fail in different platforms. I can fix it!

... and I couldn't `make verify` it here due to different linter version/behavior. I can do it now and get Static Analysis on track. 

Thanks a lot! 🙇🏼

Reply:
Please resolve the conversation if you think this has been addressed 🙇🏼